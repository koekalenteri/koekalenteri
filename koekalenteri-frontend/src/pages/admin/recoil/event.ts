import { EventEx } from "koekalenteri-shared/model"
import { atom, DefaultValue, selector, selectorFamily } from "recoil"

import { getEvent, getEvents } from "../../../api/event"
import { unique, uniqueDate } from "../../../utils"
import { logEffect, storageEffect } from "../../recoil/effects"

export interface DecoratedEvent extends EventEx {
  uniqueClasses: string[]
  uniqueClassDates: (eventClass: string) => Date[]
}

export const adminEventIdAtom = atom<string | undefined>({
  key: 'adminEventId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const currentAdminEvent = selector({
  key: 'currentAdminEvent',
  get: ({ get }) => {
    const eventId = get(adminEventIdAtom)
    return eventId ? get(adminEventSelector(eventId)) : undefined
  },
  set: ({ set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    set(adminEventSelector(newValue.id), newValue)
  },
})

function decorateEvent(event: EventEx): DecoratedEvent {
  return {
    ...event,
    uniqueClasses: unique(event.classes.map(c => c.class)),
    uniqueClassDates: (eventClass: string) => uniqueDate(
      event.classes
        .filter(c => c.class === eventClass)
        .map(c => c.date || event.startDate || new Date()),
    ),
  }
}

export const adminEventsAtom = atom<DecoratedEvent[]>({
  key: 'adminEvents',
  default: getEvents().then(events => events.map(decorateEvent)),
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const adminEventByIdAtom = selectorFamily<DecoratedEvent | undefined, string>({
  key: 'adminEvents/eventId',
  get: (eventId) => async ({ get }) => {
    let event = get(adminEventsAtom).find(event => event.id === eventId)
    if (!event) {
      const fetched = await getEvent(eventId)
      if (fetched) {
        event = decorateEvent(fetched)
      }
    }
    return event
  },
  set: (eventId) => ({ set }, newValue) => {
    if (!newValue || newValue instanceof DefaultValue) {
      return
    }
    set(adminEventsAtom, oldEvents => {
      const index = oldEvents.findIndex(item => item.id === eventId)
      const newEvents = oldEvents.map(event => ({...event}))
      newEvents.splice(index, 1, newValue)
      return newEvents
    })
  },
})

export const adminEventSelector = selectorFamily<DecoratedEvent | undefined, string>({
  key: 'adminEvent',
  get: (eventId) => ({get}) => get(adminEventByIdAtom(eventId)),
  set: (eventId) => ({set}, newValue) => set(adminEventByIdAtom(eventId), newValue),
})

export const eventClassAtom = atom<string | undefined>({
  key: 'eventClass',
  default: selector({
    key: 'eventClass/default',
    get: ({ get }) => get(currentAdminEvent)?.uniqueClasses?.[0],
  }),
  effects: [
    logEffect,
    storageEffect,
  ],
})
