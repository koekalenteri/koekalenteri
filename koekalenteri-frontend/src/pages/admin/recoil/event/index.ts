export * from "./currentEvent"
export * from "./eventClassAtom"
export * from "./eventSelector"
