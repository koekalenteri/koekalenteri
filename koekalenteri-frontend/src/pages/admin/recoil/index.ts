export * from './emailTemplates'
export * from './events'
export * from './officials'
export * from './organizers'
export * from './registrations'
