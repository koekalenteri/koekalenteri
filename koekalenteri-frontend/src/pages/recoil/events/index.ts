export * from './atoms'
export * from './effects'
export * from './filters'
export * from './selectors'
