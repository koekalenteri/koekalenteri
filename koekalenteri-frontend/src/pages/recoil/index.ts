export * from './effects'
export * from './events'
export * from './eventTypes'
export * from './judges'
export * from './officials'
export * from './organizers'
export * from './registration'
export * from './user'
