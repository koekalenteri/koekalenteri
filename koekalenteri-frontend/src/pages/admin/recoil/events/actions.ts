import { useTranslation } from "react-i18next"
import { useNavigate } from "react-router-dom"
import { useAuthenticator } from "@aws-amplify/ui-react"
import { Event } from "koekalenteri-shared/model"
import { useSnackbar } from "notistack"
import { useRecoilState, useSetRecoilState } from "recoil"

import { putEvent } from "../../../../api/event"
import { Path } from "../../../../routeConfig"

import { adminEventIdAtom, newEventAtom } from "./atoms"
import { currentAdminEventSelector } from "./selectors"


export const useAdminEventActions = () => {
  const { user } = useAuthenticator(context => [context.user])
  const setAdminEventId = useSetRecoilState(adminEventIdAtom)
  const [currentAdminEvent, setCurrentAdminEvent] = useRecoilState(currentAdminEventSelector)
  const setNewEvent = useSetRecoilState(newEventAtom)
  const { enqueueSnackbar } = useSnackbar()
  const { t } = useTranslation()
  const navigate = useNavigate()

  return {
    copyCurrent,
    deleteCurrent,
    save,
  }

  function copyCurrent() {
    if (!currentAdminEvent) {
      return
    }
    const copy = structuredClone(currentAdminEvent)
    copy.id = ''
    copy.state = 'draft'
    copy.entries = 0
    copy.classes.forEach(c => c.entries = c.members = 0)
    delete copy.kcId

    setNewEvent(copy)
    navigate(Path.admin.newEvent)
  }

  async function save(event: Partial<Event>): Promise<Event> {
    const saved = await putEvent(event, user.getSignInUserSession()?.getIdToken().getJwtToken())
    setAdminEventId(saved.id)
    setCurrentAdminEvent(saved)
    return saved
  }

  async function deleteCurrent() {
    if (!currentAdminEvent || currentAdminEvent.deletedAt) {
      return
    }

    await save({
      ...currentAdminEvent,
      deletedAt: new Date(),
      deletedBy: user.attributes?.name || user.attributes?.email,
    })

    enqueueSnackbar(t('deleteEventComplete'), { variant: 'info' })
  }

}
