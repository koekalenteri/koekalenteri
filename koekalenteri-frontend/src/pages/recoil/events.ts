
import i18next from 'i18next'
import { EventEx, Organizer } from 'koekalenteri-shared/model'
import { atom, selector, selectorFamily } from 'recoil'
import { syncEffect } from 'recoil-sync'

import { getEvent, getEvents } from '../../api/event'
import { unique, uniqueFn } from '../../utils'

import { logEffect, storageEffect } from './effects'
import { judgesAtom } from './judges'

export const eventsAtom = atom<EventEx[]>({
  key: 'events',
  default: [],
  effects: [
    logEffect,
    storageEffect,
    ({setSelf, trigger}) => {
      if (trigger === 'get') {
        getEvents().then(setSelf)
      }
    },
  ],
})

export const eventIdAtom = atom<string | undefined>({
  key: 'eventId',
  default: undefined,
  effects: [
    logEffect,
    storageEffect,
  ],
})

export const eventByIdAtom = selectorFamily<EventEx | undefined, string>({
  key: 'adminEvents/eventId',
  get: (eventId) => ({ get }) => get(eventsAtom).find(event => event.id === eventId) ?? getEvent(eventId),
})

export const eventSelector = selectorFamily<EventEx | undefined, string>({
  key: 'event',
  get: (eventId) => ({ get }) => get(eventByIdAtom(eventId)),
})

export const currentEvent = selector({
  key: 'currentEvent',
  get: ({ get }) => {
    const eventId = get(eventIdAtom)
    return eventId ? get(eventSelector(eventId)) : undefined
  },
})

export const filterJudgesQuery = selector({
  key: 'filterJudges',
  get: ({ get }) => {
    const events = get(filteredEvents)
    const judges = get(judgesAtom)
    const usedJudgeIds = unique<number>(events.reduce<number[]>((acc, cur) => [...acc, ...cur.judges], []))

    return judges
      .filter(j => usedJudgeIds.includes(j.id))
      .sort((a, b) => a.name.localeCompare(b.name, i18next.language))
  },
})

export const filterOrganizersQuery = selector({
  key: 'filterOrganizers',
  get: ({ get }) => {
    const events = get(filteredEvents)
    const uniqueOrganizesrs = uniqueFn<Organizer>(events.map(e => e.organizer), (a, b) => a.id === b.id)
    uniqueOrganizesrs.sort((a, b) => a.name.localeCompare(b.name, i18next.language))
    return uniqueOrganizesrs
  },
})

export const eventFilterAtom = atom<FilterProps>({
  key: 'eventFilter',
  default: {
    start: null,
    end: null,
    withOpenEntry: false,
    withUpcomingEntry: false,
    withClosingEntry: false,
    withFreePlaces: false,
    eventType: [],
    eventClass: [],
    judge: [],
    organizer: [],
  },
  effects: [
    logEffect,
    syncEffect({ refine: (v) => ({ type: 'success', value: v, warnings: []}) }),
  ],
})

export const filteredEvents = selector({
  key: 'filteredEvents',
  get: ({get}) => {
    const filter = get(eventFilterAtom)
    const events = get(eventsAtom)

    return events.filter(event => {
      return event.state !== 'draft' && !event.deletedAt
        && withinDateFilters(event, filter)
        && withinSwitchFilters(event, filter)
        && withinArrayFilters(event, filter)
    })
  },
})

export type FilterProps = {
  start: Date | null
  end: Date | null
  withOpenEntry?: boolean
  withClosingEntry?: boolean
  withUpcomingEntry?: boolean
  withFreePlaces?: boolean
  eventType: string[]
  eventClass: string[]
  judge: number[]
  organizer: number[]
}

function withinDateFilters(event: EventEx, { start, end }: FilterProps) {
  if (start && (!event.endDate || event.endDate < start)) {
    return false
  }
  if (end && (!event.startDate || event.startDate > end)) {
    return false
  }
  return true
}

function withinSwitchFilters(event: EventEx, { withOpenEntry, withClosingEntry, withUpcomingEntry, withFreePlaces }: FilterProps) {
  let result

  if (withOpenEntry) {
    result =  event.isEntryOpen
    if (withClosingEntry) {
      result = result && event.isEntryClosing
    }
    if (withFreePlaces) {
      result = result && event.places > event.entries
    }
  }

  if (withUpcomingEntry) {
    result = result || event.isEntryUpcoming
  }

  return result !== false
}

function withinArrayFilters(event: EventEx, { eventType, eventClass, judge, organizer }: FilterProps) {
  if (eventType.length && !eventType.includes(event.eventType)) {
    return false
  }
  if (eventClass.length && !eventClass.some(c => event.classes.map(cl => cl.class).includes(c))) {
    return false
  }
  if (judge.length && !judge.some(j => event.judges?.includes(j))) {
    return false
  }
  if (organizer.length && !organizer.includes(event.organizer?.id)) {
    return false
  }
  return true
}
