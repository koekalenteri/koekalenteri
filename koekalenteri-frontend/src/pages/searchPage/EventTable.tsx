import { useCallback } from 'react'
import { useTranslation } from 'react-i18next'
import { KeyboardArrowDown, KeyboardArrowRight } from '@mui/icons-material'
import {
  Box,
  Collapse,
  Grid,
  IconButton,
  Paper,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableRow,
} from '@mui/material'
import type { Event, EventState } from 'koekalenteri-shared/model'
import { useRecoilState } from 'recoil'

import { isEntryOpen } from '../../utils'
import LinkButton from '../components/LinkButton'
import { openedEventAtom } from '../recoil'

import { EventInfo } from './EventInfo'

function eventClasses(event: Event) {
  const ret: string[] = []
  for (const c of event.classes) {
    const name = typeof c === 'string' ? c : c.class
    if (ret.indexOf(name) === -1) {
      ret.push(name)
    }
  }
  return ret.join(', ')
}

const Row = ({ event }: { event: Event }) => {
  const [open, setOpen] = useRecoilState(openedEventAtom(event.id))
  const { t } = useTranslation()

  const handleClick = useCallback(() => setOpen(!open), [open, setOpen])

  return (
    <>
      <TableRow
        sx={{
          '& > td': {
            backgroundColor: 'background.form',
            borderBottom: '2px solid white',
            borderRadius: 4,
            padding: '2px 0',
            whiteSpace: 'nowrap',
          },
          '& div.MuiGrid-item': {
            overflow: 'hidden',
          },
          '&:last-child td': { border: 0 },
        }}
      >
        <TableCell>
          <Grid container spacing={0} alignItems="center">
            <Grid item xs={'auto'}>
              <IconButton aria-label="expand row" size="small" color="primary" onClick={handleClick}>
                {open ? <KeyboardArrowDown /> : <KeyboardArrowRight />}
              </IconButton>
            </Grid>
            <Grid item container xs onClick={handleClick}>
              <Grid item container xs={12} md={6} justifyContent="flex-start" spacing={1}>
                <Grid item xs={3} sx={{ fontWeight: event.entryOrigEndDate ? 'bold' : 'normal' }}>
                  {t('daterange', { start: event.startDate, end: event.endDate })}
                </Grid>
                <Grid item xs={2}>
                  {event.eventType}
                </Grid>
                <Grid item xs={2}>
                  {eventClasses(event)}
                </Grid>
                <Grid item xs={5}>
                  {event.location}
                  {event.name ? ` (${event.name})` : ''}
                </Grid>
              </Grid>
              <Grid item container xs={12} md={6} spacing={1}>
                <Grid item xs={6} md={7}>
                  {event.organizer?.name}
                </Grid>
                <Grid item xs={3} md={2}>
                  <EventPlaces event={event} />
                </Grid>
                <Grid item xs={3} md={3} textAlign="right">
                  {isEntryOpen(event) ? (
                    <LinkButton to={`/event/${event.eventType}/${event.id}`} text={t('register')} />
                  ) : (
                    <EventStateInfo state={event.state} />
                  )}
                </Grid>
              </Grid>
            </Grid>
          </Grid>
          <Collapse
            in={open}
            sx={{
              borderTop: '1px solid #BDBDBD',
              ml: '34px',
              mt: 1,
              pt: 1,
            }}
            timeout="auto"
            unmountOnExit
          >
            <EventInfo event={event}></EventInfo>
          </Collapse>
        </TableCell>
      </TableRow>
    </>
  )
}

const EventPlaces = ({ event }: { event: Event }) => {
  const { t } = useTranslation()
  const color = (event.entries ?? 0) > event.places ? 'warning.main' : 'text.primary'
  let text = ''
  let bold = false
  if (event.places) {
    if (event.entries) {
      text = `${event.entries} / ${event.places}`
      bold = true
    } else {
      text = event.places + ' ' + t('toltaPlaces')
    }
  }
  return (
    <Box textAlign="right" sx={{ color, fontWeight: bold ? 'bold' : 'normal' }}>
      {text}
    </Box>
  )
}

function EventStateInfo({ state }: { state: EventState }) {
  const { t } = useTranslation()
  const showInfo = state === 'tentative' || state === 'cancelled'
  return (
    <Box sx={{ color: 'warning.main', textTransform: 'uppercase', mr: 1 }}>
      {showInfo && t(`event.states.${state}_info`)}
    </Box>
  )
}

function EmptyResult() {
  const { t } = useTranslation()
  return <Box sx={{ width: '100%', textAlign: 'center', color: 'red' }}>{t('noResults')}</Box>
}

export function EventTable({ events }: { events: Event[] }) {
  return (
    <>
      {events.length ? (
        <TableContainer component={Paper}>
          <Table aria-label="event table">
            <TableBody>
              {events.map((event) => (
                <Row key={event.id} event={event} />
              ))}
            </TableBody>
          </Table>
        </TableContainer>
      ) : (
        <EmptyResult />
      )}
    </>
  )
}
