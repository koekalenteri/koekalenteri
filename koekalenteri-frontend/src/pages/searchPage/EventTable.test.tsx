import { MemoryRouter } from 'react-router-dom'
import { ThemeProvider } from '@mui/material'
import { render, screen } from '@testing-library/react'
import { parseISO } from 'date-fns'
import { Event } from 'koekalenteri-shared/model'
import { RecoilRoot } from 'recoil'

import { eventWithEntryOpen } from '../../__mockData__/events'
import { emptyEvent } from '../../api/test-utils/emptyEvent'
import theme from '../../assets/Theme'

import { EventTable } from './EventTable'

describe('EventTable', () => {
  it('should render with empty result', () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <EventTable events={[]} />
        </RecoilRoot>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('should render', async function () {
    const event: Event = {
      ...emptyEvent,
      startDate: parseISO('2021-02-10'),
      endDate: parseISO('2021-02-11'),
    }
    const { container } = render(
      <ThemeProvider theme={theme}>
        <RecoilRoot>
          <EventTable events={[event]} />
        </RecoilRoot>
      </ThemeProvider>
    )
    expect(container).toMatchSnapshot()
  })

  it('should render registration link', async function () {
    render(
      <ThemeProvider theme={theme}>
        <MemoryRouter>
          <RecoilRoot>
            <EventTable events={[eventWithEntryOpen]} />
          </RecoilRoot>
        </MemoryRouter>
      </ThemeProvider>
    )
    expect(screen.getByRole('link')).toHaveAttribute(
      'href',
      `/event/${eventWithEntryOpen.eventType}/${eventWithEntryOpen.id}`
    )
  })
})
