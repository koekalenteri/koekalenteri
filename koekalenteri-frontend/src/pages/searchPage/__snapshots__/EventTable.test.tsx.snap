// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventTable should render 1`] = `
<div>
  <div
    class="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
  >
    <table
      aria-label="event table"
      class="MuiTable-root css-166vbg3-MuiTable-root"
    >
      <tbody
        class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1lvh4ei-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container css-1vam7s3-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-auto css-kcsfla-MuiGrid-root"
              >
                <button
                  aria-label="expand row"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall css-10frfwe-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="KeyboardArrowRightIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59 13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-true css-2ryc1y-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1s2fltb-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 css-19p55ea-MuiGrid-root"
                  >
                    daterange
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1rl1j2j-MuiGrid-root"
                  >
                    test
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1rl1j2j-MuiGrid-root"
                  >
                    ALO
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-1ek8k2z-MuiGrid-root"
                  >
                    test
                     (test)
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-xn8b0u-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 MuiGrid-grid-md-7 css-rhw740-MuiGrid-root"
                  >
                    Suomen Noutajakoirajärjestö ry
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-2 css-nuhzl1-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-s2uf1z"
                    >
                      10 toltaPlaces
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3 MuiGrid-grid-md-3 css-1xd4zan-MuiGrid-root"
                  >
                    <div
                      class="MuiBox-root css-bmuutb"
                    />
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-hidden css-fhr8k5-MuiCollapse-root"
              style="min-height: 0px;"
            >
              <div
                class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
              >
                <div
                  class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                >
                  <table
                    aria-label="details"
                    class="MuiTable-root css-ii9sbb-MuiTable-root"
                  >
                    <tbody
                      class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                    >
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          scope="row"
                        >
                          entryTime
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        >
                          daterange
                          <span
                            class="info"
                          />
                        </td>
                      </tr>
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          scope="row"
                        >
                          event.classes
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        >
                          <table
                            class="MuiTable-root css-1k2pdqd-MuiTable-root"
                          >
                            <tbody
                              class="MuiTableBody-root css-apqrd9-MuiTableBody-root"
                            >
                              <tr
                                class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                              >
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                                  scope="row"
                                >
                                  eeeeee d.M.
                                </th>
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                                  scope="row"
                                >
                                  ALO
                                </th>
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                                  scope="row"
                                />
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                                  scope="row"
                                />
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight MuiTableCell-sizeSmall css-dv0mq-MuiTableCell-root"
                                  scope="row"
                                />
                                <th
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                                  scope="row"
                                />
                                <td
                                  class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                                />
                              </tr>
                            </tbody>
                          </table>
                        </td>
                      </tr>
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          rowspan="1"
                          scope="row"
                        >
                          event.judges
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        />
                      </tr>
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          scope="row"
                        >
                          event.official
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        >
                          Teemu Toimitsija
                        </td>
                      </tr>
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          scope="row"
                        >
                          event.paymentDetails
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        >
                          event.cost
                          : 
                          123
                           €
                          <br />
                          event.costMember
                          : 
                          123
                           €
                          <br />
                          <br />
                          test
                        </td>
                      </tr>
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          scope="row"
                        >
                          event.priority
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        >
                          <div
                            class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorDefault MuiChip-filledDefault css-1xzzkbc-MuiChip-root"
                          >
                            <span
                              class="MuiChip-label MuiChip-labelSmall css-wjsjww-MuiChip-label"
                            >
                              Jäsenet
                            </span>
                          </div>
                        </td>
                      </tr>
                      <tr
                        class="MuiTableRow-root css-1gsj6qz-MuiTableRow-root"
                      >
                        <th
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                          scope="row"
                        >
                          event.description
                          :
                        </th>
                        <td
                          class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall css-dsuxgy-MuiTableCell-root"
                        >
                          test
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`EventTable should render with empty result 1`] = `
<div>
  <div
    class="MuiBox-root css-fzwid6"
  >
    noResults
  </div>
</div>
`;
